### PR TITLE
feat: [BB-243] FEATURE** formik for fertilizers

### PR DESCRIPTION
**Pull request by kcaparas1630**
# Description

Implemented Formik for the fertilizers form module, and created a state hook in Parent Component to pass in the data that has been entered in the input fields. These data will then be used for the calculation module when adding fertilizers.
State hook in both components is an array of objects. Child Component uses forEach to pass in the data one by one.

Wireframe Reference:
Group 160
Group 162
Group 164
Group 165
Group 166
Group 167
Group 156
Group 159

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-270-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-270-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

**Review by dallascrichmond — APPROVED** [review]
There's nothing significant that I can see that needs to be changed. Everything seems to run as intended locally. Good job

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ const CropsInfoComponent: React.FC<InputModuleProps> = ({
   farmDetails,
   updateFarmDetails,
   handleFormState,
+  updateNutrientDetails,
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx:48]: Why is this getting passed here?

File: frontend/src/Interface/InputModuleProps.tsx [hunk 2]
```diff
@@ -7,10 +7,12 @@
  * @author    @GDamaso
  */
 import FarmDetailsInterface from './FarmDetailsInterface';
+import { TempNutrientsInterface } from './NutrientsInterface';
 
 interface InputModuleProps {
   farmDetails: FarmDetailsInterface;
   updateFarmDetails: (farmDetails: FarmDetailsInterface) => void;
+  updateNutrientDetails: (nutrientDetails: TempNutrientsInterface) => void;
```
Comment (GDamaso) [inline frontend/src/Interface/InputModuleProps.tsx:15]: This should be optional, only fertilizers and calculation need it

File: frontend/src/Views/MainPage/MainPage.tsx [hunk 3]
```diff
@@ -17,6 +17,8 @@ import { ACTIVE, COMPLETED, WARNING } from '@Constants/ModuleStatus';
 import CmdOptions from '@Constants/CmdOptions';
 import Names from '@Constants/Names';
 import convertToNMP from '@Utils/convertToNMP';
+import tempNutrientsStorage from '@Constants/TempStorageNutrients';
```
Comment (GDamaso) [inline frontend/src/Views/MainPage/MainPage.tsx:20]: should the import match the file name?

File: frontend/src/Views/MainPage/MainPage.tsx [hunk 4]
```diff
@@ -173,12 +176,28 @@ const MainPage: FC = () => {
...
+  const updateNutrientDetails = (newNutrients: TempNutrientsInterface): void => {
+    if (!nutrientsSliceFirstUpdate) {
+      nutrientsSliceFirstUpdate = true;
+      setNutrientDetails((prevDetails) => [...prevDetails.slice(1), newNutrients]);
+    } else {
+      setNutrientDetails((prevDetails) => [...prevDetails, newNutrients]);
+    }
+  };
+  // Development Log
+  console.log(nutrientDetails);
```
Comment (GDamaso) [inline frontend/src/Views/MainPage/MainPage.tsx:200]: Is this staying here?

**Review by GDamaso — COMMENTED** [review]
Good job mate, just got a couple comments :smile:

File: frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx (hunk 1 above)
Comment (kcaparas1630) [reply frontend/src/Commons/Forms/InputModules/Crops/CropsInfo.tsx]: There are two ways in going in here. If updateNutrientDetails is required, and CropsList is extending InputModuleProps. I would need to pass in that updateNutrientDetails even if I don't use it on CropsList.
Second way is to making it optional and then testing it for validity.

Either way works. but made it optional now, and tested for validity.

File: frontend/src/Interface/InputModuleProps.tsx (hunk 2 above)
Comment (kcaparas1630) [reply frontend/src/Interface/InputModuleProps.tsx]: My answer is from the previous comment.

File: frontend/src/Views/MainPage/MainPage.tsx (hunk 4 above)
Comment (kcaparas1630) [reply frontend/src/Views/MainPage/MainPage.tsx:200]: This is staying until I make the List. This is for the purpose of proving that I am able to pass in the data from the module into the parent component.

**Pull request merged**